op-challenger/game/fault/contracts/disputegame.go

```diff
@@ -13,64 +13,33 @@ import (
 )
 
 const (
-	methodGameDurationV0     = "GAME_DURATION"
-	methodMaxGameDepthV0     = "MAX_GAME_DEPTH"
-	methodAbsolutePrestateV0 = "ABSOLUTE_PRESTATE"
-	methodGameDurationV1     = "gameDuration"
-	methodMaxGameDepthV1     = "maxGameDepth"
-	methodAbsolutePrestateV1 = "absolutePrestate"
-	methodStatus             = "status"
-	methodClaimCount         = "claimDataLen"
-	methodClaim              = "claimData"
-	methodL1Head             = "l1Head"
-	methodResolve            = "resolve"
-	methodResolveClaim       = "resolveClaim"
-	methodAttack             = "attack"
-	methodDefend             = "defend"
-	methodStep               = "step"
-	methodAddLocalData       = "addLocalData"
-	methodVMV0               = "VM"
-	methodVMV1               = "vm"
+	methodGameDuration     = "gameDuration"
+	methodMaxGameDepth     = "maxGameDepth"
+	methodAbsolutePrestate = "absolutePrestate"
+	methodStatus           = "status"
+	methodClaimCount       = "claimDataLen"
+	methodClaim            = "claimData"
+	methodL1Head           = "l1Head"
+	methodResolve          = "resolve"
+	methodResolveClaim     = "resolveClaim"
+	methodAttack           = "attack"
+	methodDefend           = "defend"
+	methodStep             = "step"
+	methodAddLocalData     = "addLocalData"
+	methodVM               = "vm"
 )
 
 type disputeGameContract struct {
 	multiCaller *batching.MultiCaller
 	contract    *batching.BoundContract
-	// The version byte signifies the version of the dispute game contract due to mismatching function selectors.
-	// 0 = `FaultDisputeGame`
-	// 1 = `OutputBisectionGame`
-	version uint8
-}
-
-// contractProposal matches the structure for output root proposals used by the contracts.
-// It must exactly match the contract structure. The exposed API uses Proposal to decouple the contract
-// and challenger representations of the proposal data.
-type contractProposal struct {
-	Index         *big.Int
-	L2BlockNumber *big.Int
-	OutputRoot    common.Hash
 }
 
 type Proposal struct {
 	L2BlockNumber *big.Int
 	OutputRoot    common.Hash
 }
 
-func asProposal(p contractProposal) Proposal {
-	return Proposal{
-		L2BlockNumber: p.L2BlockNumber,
-		OutputRoot:    p.OutputRoot,
-	}
-}
-
 func (f *disputeGameContract) GetGameDuration(ctx context.Context) (uint64, error) {
-	var methodGameDuration string
-	if f.version == 1 {
-		methodGameDuration = methodGameDurationV1
-	} else {
-		methodGameDuration = methodGameDurationV0
-	}
```

Comment:
This PR is fixing it which is great, but for the record, introducing these if cases is the wrong pattern - we should have just moved the methods to the game type specific structs (`OutputBisectionGameContract` and `FaultDisputeGameContract` which build on this shared code).

Generally it's better to use "branch by abstraction" rather than having if statements spread through the code.  That said, it's not too bad in this case because it is all encapsulated here rather than spread through the actual business logic.  And I realise you knew it was going to be deleted pretty soon after. :)

Reply:
As a follow up now that we've only got one fault dispute game contract type we should just inline this "shared" code into the one remaining struct.

Reply:
Ticketed out here: https://github.com/ethereum-optimism/client-pod/issues/402